common/lib/xmodule/xmodule/contentstore/content.py

```diff
@@ -127,11 +134,100 @@ def get_location_from_path(path):
                 return AssetKey.from_string(path[1:])
 
     @staticmethod
+    def get_canonicalized_asset_path(course_key, path):
+        """
+        Returns a fully-qualified path to a piece of static content.
+
+        If a static asset CDN is configured, this path will include it.
+        Otherwise, the path will simply be relative.
+
+        Args:
+            course_key: key to the course which owns this asset
+            path: the path to said content
+
+        Returns:
+            string: fully-qualified path to asset
+        """
+
+        # Break down the input path.
+        _, base_url, relative_path, params, query_string, fragment = urlparse(path)
+
+        # Convert our path to an asset key.
+        asset_key = StaticContent.get_asset_key_from_path(course_key, relative_path)
+
+        # Check to see if we can actually provide a CDN-ified version
+        # of this asset back to the caller.
+        is_locked = False
+        try:
+            content = AssetManager.find(asset_key, as_stream=False)
+            is_locked = getattr(content, "locked", False)
+        except (ItemNotFoundError, NotFoundError):
+            # TODO: What's the right fallback here?  Assume it's locked?
+            # Can we ever reasonably except this state since we're actually
+            # calling the same store that will be asked to serve the asset?
+            pass
```

Comment:
I think the most reasonable thing to do is to log it as a warning and fall back to the non-CDN relative path that `convert_legacy_static_url_with_course_id` would have returned. There might be some weird code path that we're not seeing, but doing that will ensure that things will at least be no more broken than they are today.